setup.cfg

```diff
@@ -44,7 +44,7 @@ install_requires =
     certifi
     jsonschema>=3.0
 
-python_requires = >=3.6
+python_requires = >=3.8
```

Comment:
While I'm :+1: about changing the references elsewhere to 3.8 as the earliest supported version, I'd rather keep this hard installation constraint at 3.6. It _should_ still install and work on 3.6, since there haven't been functional changes.

Sure, no one _should_ be using 3.6 or 3.7 any more, but odds are someone somewhere _is_ stuck on an EOL version. Might as well not block them from installing a new sphobjinv version.

Reply:
How would you ensure that the package still works for someone using 3.6 or 3.7 unless you're also testing 3.6 and 3.7? See, I have myself been in the unfortunate situation of having to support a Python version after it goes EOL, and my experience is that once you hit EOL it suddenly gets much harder, mostly due to PyPI. People keep pushing changes and releasing new versions, but they stop testing against the old version of Python. Now, if everyone who stopped testing an old Python version updated their `python_requires`, I would have no problem — a `pip install` of my package would just get older versions of my dependencies, and everything that was working should keep working. In practice, too many people stop testing without updating their `python_requires` and supporting the older version becomes a dependency nightmare, where you have to specify a latest supported version for not just your direct dependencies, but all the transitive dependencies as well.

If there's new functionality that we're keen on letting the 3.6 and 3.7 users have, then let's please keep testing that it works with 3.6 and 3.7 (and be prepared to take more responsibility for your dependencies than you might otherwise have to). If, on the other hand, we think that the functionality of the already released versions of sphobjinv is still sufficient for their expected uses, then we're not doing them any harm by raising our `python_requires` to be in line with what we're actually testing.

Reply:
Ah, interesting, I had been focused on direct users of `sphobjinv`, and not on packages that have `sphobjinv` as a transitive dependency... I was seeing the argument for leaving the `python_requires` at a version below what's actively tested as something of the flip-side of the argument for not including max-version constraints in dependency specifications... better to allow the possibility of a version working for someone, even if some users then have to add an extra pin.

But your point about the mess of transitive dependencies is well taken. I still have Python 3.7 on my local machine that I run the tox matrix on, but I usually only run that on Linux, not Windows.

So, you've convinced me on this `python_requires` bump. Thanks for pushing back!